inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -45,13 +45,41 @@ class WPSEO_Post_Type_Sitemap_Provider implements WPSEO_Sitemap_Provider {
 	 */
 	protected static $page_for_posts_id;
 
+	/**
+	 * The with_front rewrite setting.
+	 *
+	 * @var boolean
+	 */
+	private $post_type_has_front = true;
+
 	/**
 	 * Set up object properties for data reuse.
 	 */
 	public function __construct() {
 		add_filter( 'save_post', array( $this, 'save_post' ) );
 	}
 
+	/**
+	 * Sets the post type has front boolean.
+	 *
+	 * @param string $post_type the post type.
+	 *
+	 * @return void
+	 */
+	protected function set_post_type_has_front( $post_type ) {
+		$post_type_has_front = get_post_type_object( $post_type )->rewrite['with_front'];
```

Comment:
It's possible that `rewrite['with_front']` is not set. Also, we should check is it working in PHP 5.2.4. You should improve "code readability" ( I think that Yoast codebase doesn't use function dereferencing).

Reply:
Updated in latest commit.